lib/credentials/basicAuthenticationCredentials.ts

```diff
@@ -44,4 +45,17 @@ export class BasicAuthenticationCredentials implements ServiceClientCredentials
     webResource.headers[HeaderConstants.AUTHORIZATION] = encodedCredentials;
     return Promise.resolve(webResource);
   }
+
+  /**
+   * Signs a request with the Authentication header.
+   *
+   * @param httpRequest The request to be signed.
+   * @returns The signed request.
+   */
+  public signHttpRequest(httpRequest: HttpRequest): Promise<HttpRequest> {
```

Comment:
This is a temporary change that will be reverted when I change WebResource to HttpRequest. I could've done method overloading here, but rather than learn how to do that in TypeScript, I figured I'd just use a different name for now and then switch it back to signRequest(HttpRequest) when I change to using HttpRequest instead of WebResource.
Just to be clear, changing to use the new HttpPipeline will break customers. This is going to be a breaking change. I understand that it's a good idea to minimize the number of breaking changes, but it's not completely avoidable.

Reply:
I see. Sounds good to me. It is always good to avoid breaking changes wherever possible, unless they are absolutely required.